File: espresso-tracker/src/models/EspressoSetting.ts
import { Coffee } from "../models/Coffee";
import { CoffeeTool } from "../services/CoffeeToolsService";

export interface EspressoSetting {
  id: number;
  coffee: Coffee;
  coffeeMachine: CoffeeTool;
  grinder: CoffeeTool;
  dose: number;
  grindingFineness: string;
  brewingTemperature: number;
  brewingPressure: number;
  volume: number;
  extractDurationSec: number;
}

export interface EspressoSettingRequest {
  coffeeId: number;
  dose: number;
  grindingFineness: string;
  brewingTemperature: number;
  brewingPressure: number;
  volume: number;
  extractDurationSec: number;
}

File: espresso-tracker-service/src/main/java/com/softuni/espresso/tracker/model/EspressoSetting.java
package com.softuni.espresso.tracker.model;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class EspressoSetting {
    private long id;
    private Coffee coffee;
    private CoffeeTool coffeeMachine;
    private CoffeeTool grinder;
    private double dose;
    private String grindingFineness;
    private double brewingTemperature;
    private double brewingPressure;
    private double volume;
    private int extractDurationSec;
}
